### ISSUE TITLE
Bootbox 5.0 ship list

### ISSUE THREAD

**Issue by tiesont** [makeusabrew/bootbox#665]
- [x] Generate minified file for bootbox.js
- [x] Generate minified file for bootbox.locales.js
- [x] Generate minified file for bootbox.all.js (combined bootbox.js and bootbox.locales.js)
- [x] Generate zip archive for Releases tab
- [ ] Update CDNjs
- [x] Update NPM (?)
- <s>Update bower (?)</s>
- [x] Update NuGet
- [x] Update README to note locales being pulled to separate file
- [x] Update documentation site for new and updated features

**Comment by tiesont** [comment]
@tarlepp I can update the NuGet package, and I _think_ CDNjs updates itself based on the repo being updated, but do you know how bower and/or NPM are updated? I was able to do a clean pull-request from v5.x to master on my fork, so I think that's all I need to do at this point to go live with version 5.  

I have a draft of a 5.0 release setup, so I just need to update the files there to the most recent updates, and that will be squared away.

Would be nice to definitively close the "doesn't work with BS4" issues.

**Comment by meglio** [comment]
So, is there a way to npm-install the 5.0 version? Is it already stable enough to be used in production?

**Comment by tiesont** [comment]
I don't use non, but I don't think so. I'd have to read up on how that gets updated, which is one reason this hasn't been pushed out yet.

I've used this in at least one site that went to production, so I would say yes. There really aren't any code changes I plan on making - I just need to get all the external updates squared away so that everything is in sync.

**Comment by kounelios13** [comment]
@tiesont As far as I know only the owner of the npm package can update it on the NPM repo

**Comment by lddubeau** [comment]
@tiesont Did you consider forking the project? I mean spawning a new project derived from this one, under a new name, with you as owner? Looking at the profile of the project owner, I don't see any activity on github in 2018, almost nothing in 2017, with the last year of substantial activity on github being 2016.

**Comment by tiesont** [comment]
@lddubeau Considered, yes, but I'd rather keep this project going, if possible.

**Comment by nash-ye** [comment]
We are using this library for awhile, what left to push the support for Bootstrap v4?

**Comment by tiesont** [comment]
@nash-ye Depends on what kind of support you mean. I can push my updates into the master branch, which would correct the templates for both BS3 and BS4 support, and add a few new features. Neither I nor @tarlepp can update the external resources like npm, bower, etc., nor can we publish new releases. So, we haven't updated master until doing the aforementioned tasks is possible.

**Comment by peterfarrell** [comment]
@tiesont Could you update master anyways? I don't use this project via NPM or bower so that seems like a different issue that shouldn't hold up an update.

**Comment by nash-ye** [comment]
@tiesont I agree with @peterfarrell . If it's impossible to update NPM, Bower..etc, I would suggest to create another repository or to deprecate the packages in the 3rd-party services. This library is very helpful and many projects are switching to BS4.

**Comment by lddubeau** [comment]
As I suggested a [while back](https://github.com/makeusabrew/bootbox/issues/665#issuecomment-456503101), I've [forked](https://github.com/lddubeau/bootprompt) bootbox. The fork includes tiesont's work for BS4 support. Each new version is pushed to [npm](https://www.npmjs.com/package/bootprompt). The latest release is still "alpha" because there were a few surprises during the fork that still need to be addressed. I'm nonetheless using it in production. (Then again I don't use all features so YMMV.) Folks who want to use it should pay special attention to [this document](https://github.com/lddubeau/bootprompt/blob/master/docs/bootbox-to-bootprompt.md).

**Comment by tiesont** [comment]
@peterfarrell @nash-ye I'll see what I can do, but it probably won't be until this weekend at the earliest. As @lddubeau has stated, he's created a fork that might be a better option going forward if you need npm support.

**Comment by tiesont** [comment]
Okay, @lddubeau's work inspired me to see if I could correct things like the broken tests and whatnot. It did require me to install npm, which I didn't really want to do (we don't use node or npm at work), but after about a day and a half of work, I think everything I _can_ get squared away, is. This was all done on my fork, so I'll be working on pushing my changes into master here and then pushing a new item into the Releases tab. Once that's done, I don't see myself making much in the way of updates, since I like the work @lddubeau has done on his version (especially if you're looking for a TypeScript version).

So, be on the lookout for a 5.0 release sometime this week.

**Comment by tiesont** [comment]
Okay, folks - I went ahead and pushed out a 5.0 release. I am updating the NuGet package as I write this, but I have no idea if/when CDNjs, npm, or bower will be updated. I _think_ CDNjs auto-updates from the contents of the repository, so that should take care of itself. If npm and bower do the same, great. Otherwise, unless/until @makeusabrew pops his head in, those will not be updated.

**Comment by t-veron** [comment]
For npm you must publish your package with the command « npm-publish »

Please can you do it ?

https://docs.npmjs.com/updating-your-published-package-version-number

https://eladnava.com/publishing-your-first-package-to-npm/

**Comment by tiesont** [comment]
@t-veron No, I can't. @makeusabrew is the owner of that package, and he hasn't been responding.

**Comment by tiesont** [comment]
@max-ci Yes. 5.0.0 is the current master, pushed yesterday.

**Comment by tiesont** [comment]
There are actually two owners listed on the npm package, so I sent an email to the other owner to see if I can be added there. If so, I'll publish this to npm, which will pretty much leave CDNjs as the one thing to fix (bower seems like it's going away, so not sure it's worth the effort to figure that one out). I'll update the docs with the workaround mentioned earlier, in the meantime.

**Comment by tiesont** [comment]
I've been granted admin rights on the npm package, so I'll try to get that squared away soon.

**Comment by tiesont** [comment]
Okay, npm should be updated. I'm not super familiar with npm, so I accidentally published everything and the kitchen sink to 5.0.0 (which I assume most of you don't want). So, unpublished that and published 5.0.1 that has the contents of the /dist and /src directories (in case someone wants the un-minified versions). Please let me know if it still seems broken.

**Comment by tiesont** [comment]
Okay, cdnjs should be updated soon, then we can all rejoice in the awesomeness of this long-awaited release (I hear there is cake!). Kidding. But seriously, cdnjs is being updated. Once that's done, I'll update the docs and then close this issue.

Thank you for being patient.

**Comment by tiesont** [comment]
Still waiting on cdnjs. Not sure when they'll get around to pulling this version in. In the meanwhile, you've a new locale option, Tamil, thanks to Kolappan Nathan. Thanks!

**Comment by ray73864** [comment]
I'm not sure if this is an oversight or maybe i'm just missing something, but for some reason I can't use 'centerVertical' on a prompt.

It works 100% on dialogs and pretty sure alert boxes, but I can't get it to work on prompts at all.

I'm using it by doing:

```
bootbox.prompt({
  centerVertical: true,
  title: "My Test Title",
  callback: function(result) {  }
}).modal("show");
```

**Comment by tiesont** [comment]
@ray73864 Please open a new issue, rather than posting that here.
